typing/persistent_env.ml

```diff
@@ -864,11 +864,32 @@ let local_ident penv modname =
   | Some { ps_binding = Constant _; _ }
   | None -> None
 
-let runtime_parameters ({persistent_structures; _} as penv) =
+let is_imported_parameter penv modname =
+  match find_info_in_cache penv modname with
+  | Some pers_struct -> pers_struct.ps_name_info.pn_import.imp_is_param
+  | None -> false
+
+let runtime_parameters {persistent_structures; _} =
+  (* This over-approximates the runtime parameters that are actually needed:
+     some modules get looked at during type checking but aren't relevant to
+     generated lambda code. This is increasingly true with modes and layouts: we
+     might need to check [P.t]'s layout but never end up using [P] directly, in
+     which case `P` will end up a runtime parameter that's not needed.
+
+     On the other hand, extra parameters here don't necessarily hurt much: they
+     make [-instantiate] work harder but inlining should eliminate the actual
+     runtime performance hit. If we do end up caring, probably what we need is
+     to coordinate with [Translmod] so that we're asking "what all did we access
+     during lambda generation?" rather than "what all did anyone ask about
+     ever?". *)
   persistent_structures
-  |> Hashtbl.to_seq_keys
+  |> Hashtbl.to_seq_values
   |> Seq.filter_map
-       (fun name -> local_ident penv name |> Option.map (fun id -> name, id))
+        (fun ps ->
+           match ps.ps_binding with
+           | Runtime_parameter local_ident ->
+               Some (ps.ps_name_info.pn_global, local_ident)
+           | Constant _ -> None)
```

Comment:
Might be better to move this change to the previous PR. Also, the function `local_ident` is no longer needed?

Reply:
Deleting `local_ident`.